pkg/controllers/jobset_controller.go

```diff
@@ -274,29 +274,15 @@ func (r *JobSetReconciler) executeFailurePolicy(ctx context.Context, js *jobset.
 		return r.failJobSet(ctx, js)
 	}
 
-	// Handle different types of failure policy targets.
-	switch js.Spec.FailurePolicy.Operator {
-	case jobset.TerminationPolicyTargetAny:
-		// To reach this point a job must have failed, and TerminationPolicyTargetAny applies to any job
-		// in the JobSet, so we can skip directly to executing the restart policy.
-		return r.executeRestartPolicy(ctx, js, ownedJobs)
-	default:
-		return fmt.Errorf("invalid termination policy target %s", js.Spec.FailurePolicy.Operator)
-	}
+	// To reach this point a job must have failed.
+	return r.executeRestartPolicy(ctx, js, ownedJobs)
 }
 
 func (r *JobSetReconciler) executeRestartPolicy(ctx context.Context, js *jobset.JobSet, ownedJobs *childJobs) error {
-	log := ctrl.LoggerFrom(ctx)
-
-	switch js.Spec.FailurePolicy.RestartPolicy {
-	case jobset.RestartPolicyRecreateAll:
-		return r.restartPolicyRecreateAll(ctx, js, ownedJobs)
-	case jobset.RestartPolicyNone:
-		return r.failJobSet(ctx, js)
-	default:
-		log.Error(fmt.Errorf("invalid restart policy: %s", js.Spec.FailurePolicy.RestartPolicy), "invalid restart policy, defaulting to None")
+	if js.Spec.FailurePolicy.MaxRestarts == 0 {
```

Comment:
Is this necessary? I think maybe we can skip this and call restartPolicyRecreateAll directly because the first thing it does is check if JobSet has reached max number of restarts, mark it as failed and return, so if MaxRestarts == 0 it will do this immediately.

Reply:
The message in the status and event is different, that is why I kept it.